stig: fix evaluation dropping check dependency `python3Packages.asynctest`

ShamrockLee opened the pull request:
The `python3Packages.asynctest` package is unmaintained by the upstream (see issue Martiusweb/asynctest#158), and it doesn't support Python 3.10 or later. Starting from the commit 9ba1a70f7f49ed6651c0430387bc30e325de2039 ("python313Packages.yarl: 1.13.1 -> 1.17.1"), `stig` starts to depend on `python3Packages.propcache`, and the latter requires Python 3.11 or later.

While the stig upstream is working on the transition (see issue comment https://github.com/rndusr/stig/issues/206#issuecomment-939248021), this PR fixes the evaluation of `stig` by specifying `doCheck = false` and dropping the `checkInputs`.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ShamrockLee committed:
stig: use python3Package instead of python310Packages

ShamrockLee committed:
stig: disable checks and get rid of checkInputs to resolve dependency conflicts.

The chekck dependency asynctest is unmaintained by the upstream
and only supports Python<=3.9.
With such dependency,
it is impossible to find a Python interpreter version
for stig to evaluate.

ShamrockLee commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 383465`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>stig</li>
    <li>stig.dist</li>
  </ul>
</details>

ShamrockLee commented:
According to the upstream author, [stig no longer has no tests since asynctest died](https://github.com/rndusr/stig/issues/206#issuecomment-2669636320).

Let's also drop `preCheck`.

ShamrockLee committed:
stig: remove dontUseSetuptoolsCheck specification

setuptooolsCheckHooks has been removed.

dotlambda commented on pkgs/by-name/st/stig/package.nix:
```diff
@@ -30,27 +30,7 @@ python310Packages.buildPythonApplication rec {
-    # Another failure with similar circumstances to the above
-    "test_candidates_are_sorted_case_insensitively"
-  ];
+  doCheck = false;
```
Please add a comment with the reason.

dotlambda commented on pkgs/by-name/st/stig/package.nix:
```diff
@@ -19,7 +19,7 @@ python310Packages.buildPythonApplication rec {
   };
 
-  propagatedBuildInputs = with python310Packages; [
+  propagatedBuildInputs = with python3Packages; [
```
```suggestion
  dependencies = with python3Packages; [
```

dotlambda reviewed (commented):
Please use `pyproject = true` and add `build-system`.

ShamrockLee committed:
stig: modernise

- Use "pyproject = true" and "build-system = [ setuptools ]"
- Specify run-time Python dependencies via "dependencies"

Co-authored-by: Robert Schütz <github@dotlambda.de>

ShamrockLee commented:
> Please use `pyproject = true` and add `build-system`.

Addressed.

dotlambda commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 383465`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>stig</li>
    <li>stig.dist</li>
  </ul>
</details>
